src/Backups/registerBackupEngineAzureBlobStorage.cpp

```diff
@@ -0,0 +1,134 @@
+#include "config.h"
+
+#include <Backups/BackupFactory.h>
+#include <Common/Exception.h>
+
+#if USE_AZURE_BLOB_STORAGE
+#include <Backups/BackupIO_AzureBlobStorage.h>
+#include <Storages/StorageAzureBlob.h>
+#include <Backups/BackupImpl.h>
+#include <IO/Archives/hasRegisteredArchiveFileExtension.h>
+#include <Interpreters/Context.h>
+#include <Poco/Util/AbstractConfiguration.h>
+#include <filesystem>
+#endif
+
+
+namespace DB
+{
+namespace fs = std::filesystem;
+
+namespace ErrorCodes
+{
+    extern const int BAD_ARGUMENTS;
+    extern const int SUPPORT_IS_DISABLED;
+}
+
+#if USE_AZURE_BLOB_STORAGE
+namespace
+{
+    String removeFileNameFromURL(String & url)
+    {
+        Poco::URI url2{url};
+        String path = url2.getPath();
+        size_t slash_pos = path.find_last_of('/');
+        String file_name = path.substr(slash_pos + 1);
+        path.resize(slash_pos + 1);
+        url2.setPath(path);
+        url = url2.toString();
+        return file_name;
+    }
+}
+#endif
+
+
+void registerBackupEngineAzureBlobStorage(BackupFactory & factory)
+{
+    auto creator_fn = []([[maybe_unused]] const BackupFactory::CreateParams & params) -> std::unique_ptr<IBackup>
+    {
+#if USE_AZURE_BLOB_STORAGE
+        const String & id_arg = params.backup_info.id_arg;
+        const auto & args = params.backup_info.args;
+
+        LOG_INFO(&Poco::Logger::get("registerBackupEngineAzureBlobStorage"), "Begin id_arg={} args.size={}", id_arg, args.size());
+
+        StorageAzureBlob::Configuration configuration;
+
+        if (args.size() == 4)
+        {
+            configuration.connection_url = args[0].safeGet<String>();
+            configuration.is_connection_string = true;
+
+            configuration.container =  args[1].safeGet<String>();
+            configuration.blob_path = args[2].safeGet<String>();
+            configuration.format = args[3].safeGet<String>();
+
+            LOG_TRACE(&Poco::Logger::get("registerBackupEngineAzureBlobStorage"), "configuration.connection_url = {}"
+                                                                                 "configuration.container = {}"
+                                                                                 "configuration.blob_path = {}"
+                                                                                 "configuration.format = {}",
+                                                                                 configuration.connection_url, configuration.container, configuration.blob_path, configuration.format);
+        }
```

Comment:
And what if `args.size()` is not `4`? Shouldn't we tell the user that the list of the arguments are not correct?

Reply:
updated reading arguments & displaying error message